CMakeLists.txt

```diff
@@ -188,6 +188,27 @@ set (Seastar_ALLOC_PAGE_SIZE
   STRING
   "Override the Seastar allocator page size, in bytes.")
 
+function (check_nullable_option option_name package_name)
```

Comment:
I think the name "check_nullable_option" doesn't really describe what this function is doing. What it seems to really be doing is to check whether a package exists - and allows us to fail if is explictly enabled (--enable-...) or pretend the package doesn't exist even if it does when explicitly disabled (--disable-dpdk).
Would be nice to also get a comment explaining what this function does, in addition to a better name.

Reply:
thank you for your review.

> I think the name "check_nullable_option" doesn't really describe what this function is doing. 

i knew. i tried but stopped at the 4th name . the first 3 of them were "maybe_set_option", "enable_option_if_possible", "check_option_with_package". would appreciate if if any of us could help suggest a better name.

> What it seems to really be doing is to check whether a package exists - and allows us to fail if is explictly enabled (--enable-...) or pretend the package doesn't exist even if it does when explicitly disabled (--disable-dpdk).
> Would be nice to also get a comment explaining what this function does, in addition to a better name.

quite close. but it does not pretend the package doesn't exist -- it does not set the `<PackageName>_FOUND`. that'd be more confusing than setting the `WITH_<FEATURE>` variable on behalf of user.

sure, will add a comment.

Reply:
how about `try_to_enable_option_if_not_defined()`?

Reply:
I probably completely misunderstood the code, so don't take any advice from me on the naming :-) I thought I was seeing code that checks whether or not the package is found - and then verifies this fact's interaction with the option (e.g., if the package exists but the user asked not to use it, it won't).

Reply:
yeah. that's exactly what it did in the version you commented and it still has the same behavior in the merged version.